HangFire.Autofac.Tests/AutofacJobActivatorTests.cs

```diff
@@ -178,6 +178,34 @@ public void InstanceRegisteredWith_InstancePerJob_IsDisposedOnScopeDisposal()
         }
 
         [TestMethod]
+        public void InstancePerJob_RegisteredWithExtraTags_ResolvesForJobScope()
+        {
+            var alternateLifetimeScopeTag = new object();
+            _builder.Register(c => new object()).As<object>()
+               .InstancePerBackgroundJob(alternateLifetimeScopeTag);
+            var activator = CreateActivator();
+
+            using (var scope = activator.BeginScope())
+            {
+                var instance = scope.Resolve(typeof(object));
+            }
+        }
+
+        [TestMethod]
+        public void InstancePerJob_RegisteredWithExtraTags_ResolvesForAlternateScope()
+        {
+            var alternateLifetimeScopeTag = new object();
+            _builder.Register(c => new object()).As<object>()
+               .InstancePerBackgroundJob(alternateLifetimeScopeTag);
+            var container = _builder.Build();
+
+            using (var scope = container.BeginLifetimeScope(alternateLifetimeScopeTag))
+            {
+                var instance = scope.Resolve(typeof(object));
```

Comment:
shouldn't you check if resolved object instance is the same or different object depending on scope type than another object instance you resolved?